src/fn_call.rs

```diff
@@ -93,6 +93,30 @@ pub trait EvalContextExt<'a, 'mir, 'tcx: 'a + 'mir>: crate::MiriEvalContextExt<'
                     this.write_scalar(Scalar::Ptr(ptr.with_default_tag()), dest)?;
                 }
             }
+            "calloc" => {
+                let items = this.read_scalar(args[0])?.to_usize(this)?;
+                let count = this.read_scalar(args[1])?.to_usize(this)?;
+                let size = if let Some(size) = items.checked_add(count) {
+                    size
+                } else {
+                    return err!(MachineError(format!(
+                        "calloc: overflow of items * count: {} * {}",
+                        items, count,
+                    )));
```

Comment:
You can use [`ok_or_else`](https://doc.rust-lang.org/nightly/std/option/enum.Option.html#method.ok_or_else) to make this more Rust-y.